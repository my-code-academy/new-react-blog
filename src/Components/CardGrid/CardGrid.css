.card_grid{
    /* display: grid;
    grid-column-gap: 5px;
    padding: 40px 80px;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 50px;
    grid-column-gap: 5px;
    padding: 0 10px;
    margin-left: 10%; */

    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin: 0 auto;
    width: 75%;
}
