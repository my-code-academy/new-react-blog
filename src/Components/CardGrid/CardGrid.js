import React, { Component } from 'react';
import './CardGrid.css';
import Card from '../Card/Card'

class CardGrid extends Component {
  render() {
    return (
      <div className="card_grid">
        {
          this.props.cards.map((card, index) => {
            return <Card
              date = {card.date}
              readingTime = {card.readingTime}
              title = {card.title}
              description = {card.description}
              claps = {card.claps}
              liked = {card.liked}
              imageSrc = {`./Images/${card.image}`}
            ></Card>
          })
        }
      </div>
    );
  }
}

export default CardGrid;
