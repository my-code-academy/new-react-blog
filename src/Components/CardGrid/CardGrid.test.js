import React from 'react';
import renderer from 'react-test-renderer';
import CardGrid from './CardGrid';

describe('CardGrid', () => {
  it("should render a CardGrid", () => {
    const cards= [{
      "date": "2nd Januray, 2018",
      "readingTime": "2 mins",
      "title": "The future of abstract art and the culture ...",
      "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
      "claps": 10,
      "liked": false,
      "image": "abstract.png"
  }];
    const tree = renderer.create(<CardGrid cards={cards}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});