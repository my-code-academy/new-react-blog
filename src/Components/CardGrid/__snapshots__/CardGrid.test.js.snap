// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid should render a CardGrid 1`] = `
<div
  className="card_grid"
>
  <div
    class="card"
  >
    <img
      alt="abstract"
      className="image"
      src="./Images/abstract.png"
    />
    <div
      className="container"
    >
      <div
        className="dateTime"
      >
        <span
          className="date"
        >
          2nd Januray, 2018
        </span>
        <span
          className="date"
          id="readTime"
        >
          2 mins
        </span>
      </div>
      <p
        className="title"
      >
        The future of abstract art and the culture ...
      </p>
      <p
        className="abstract"
      >
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <hr />
      <div
        className="responses"
      >
        <img
          alt="clap"
          className="clap"
          onClick={[Function]}
          src="clapping.svg"
        />
        <span>
          10
        </span>
        <img
          alt=""
          className="heart"
          onClick={[Function]}
          src="heart-red.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
